src/Microsoft.IdentityModel.Tokens/Encryption/RsaKeyWrapProvider.cs

```diff
@@ -0,0 +1,459 @@
+        public string Context { get; set; }
+
+        /// <summary>
+        /// Gets the <see cref="SecurityKey"/> that is being used.
+        /// </summary>
+        public SecurityKey Key { get; private set; }
+
+        private RSAParameters CreateRsaParametersFromJsonWebKey(JsonWebKey webKey, bool isPrivate)
+        {
+            if (webKey == null)
+                throw LogHelper.LogArgumentNullException(nameof(webKey));
+
+            if (webKey.N == null || webKey.E == null)
+                throw LogHelper.LogExceptionMessage(new ArgumentException(String.Format(CultureInfo.InvariantCulture, LogMessages.IDX10700, webKey)));
+
+            RSAParameters parameters;
+            if (isPrivate)
+            {
+                if (webKey.D == null || webKey.DP == null || webKey.DQ == null || webKey.QI == null || webKey.P == null || webKey.Q == null)
+                    throw LogHelper.LogExceptionMessage(new ArgumentNullException(nameof(webKey), String.Format(CultureInfo.InvariantCulture, LogMessages.IDX10702, webKey)));
+
+                parameters = new RSAParameters()
+                {
+                    D = Base64UrlEncoder.DecodeBytes(webKey.D),
+                    DP = Base64UrlEncoder.DecodeBytes(webKey.DP),
+                    DQ = Base64UrlEncoder.DecodeBytes(webKey.DQ),
+                    Exponent = Base64UrlEncoder.DecodeBytes(webKey.E),
+                    Modulus = Base64UrlEncoder.DecodeBytes(webKey.N),
+                    InverseQ = Base64UrlEncoder.DecodeBytes(webKey.QI),
+                    P = Base64UrlEncoder.DecodeBytes(webKey.P),
+                    Q = Base64UrlEncoder.DecodeBytes(webKey.Q)
+                };
+            }
+            else
+            {
+                parameters = new RSAParameters()
+                {
+                    Exponent = Base64UrlEncoder.DecodeBytes(webKey.E),
+                    Modulus = Base64UrlEncoder.DecodeBytes(webKey.N),
+                };
+            }
+            return parameters;
+        }
+
+        /// <summary>
+        /// Calls <see cref="Dispose(bool)"/> and <see cref="GC.SuppressFinalize"/>
+        /// </summary>
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Disposes of internal components.
+        /// </summary>
+        /// <param name="disposing">true, if called from Dispose(), false, if invoked inside a finalizer.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing)
+                {
+#if NETSTANDARD1_4
+                    if (_rsa != null && _disposeRsa)
+                        _rsa.Dispose();
+#else
+                    if (_rsaCryptoServiceProvider != null && _disposeRsa)
+                        _rsaCryptoServiceProvider.Dispose();
+#endif
+
+                    if (_rsaCryptoServiceProviderProxy != null)
```

Comment:
See comments in: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/559